modules/core/src/main/java/io/fluo/recipes/transaction/RecordingTransactionBase.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015 Fluo authors (see AUTHORS)
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.fluo.recipes.transaction;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+import io.fluo.api.client.TransactionBase;
+import io.fluo.api.config.ScannerConfiguration;
+import io.fluo.api.data.Bytes;
+import io.fluo.api.data.Column;
+import io.fluo.api.exceptions.AlreadySetException;
+import io.fluo.api.iterator.RowIterator;
+
+public class RecordingTransactionBase implements TransactionBase {
+
+  private final TransactionBase txb;
+  private TxLog txLog = new TxLog();
+
+  protected RecordingTransactionBase(TransactionBase txb) {
```

Comment:
I can't make it private because it needs to be called by `RecordingTransaction` constructor but was able to make it package-private in 59cb5a7aa07